src/main/java/com/socialmedia/util/friendship/Graph.java

```diff
@@ -0,0 +1,84 @@
+package com.socialmedia.util.friendship;
+
+import com.socialmedia.model.ApplicationUser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.Stack;
+import java.util.Queue;
+import java.util.LinkedList;
+
+class Graph {
```

Comment:
I guess, it should be called UserGraph or something like this ore refactored as generic